docs/documentation.md

```diff
@@ -0,0 +1,470 @@
+<!--
+   - SPDX-FileCopyrightText: 2022 Tezos Commons
+   -
+   - SPDX-License-Identifier: LicenseRef-MIT-TC
+   -->
+
+# Homebase Lite
+
+
```

Comment:
So... with the contract code for a `recursive` `Parameter` definition, I figured there are two options, either go with

```haskell
case_ param 
  ( #sAdmin #= adminEntrypoints
  , #sVoting #= votingEntrypoints
  )
```

In this case, autodoc says to wrap the parameter appropriately and call the default entrypoint. Not ideal, to say the least.

The other option is to use `entryCase`:


```haskell
entryCase (Proxy @PlainEntrypointsKind) param 
  ( #sAdmin #= adminEntrypoints
  , #sVoting #= votingEntrypoints
  )
```

But then `Admin` and `Voting` constructors are treated as entrypoints in their own right by autodoc. Also not ideal.

Reply:
I see, it took me a while to look into this, but I think it makes sense to keep the current choice.

Apparently this can be fixed, by using `delegate` for the outer `Parameter` (and `plain` for the inner ones) with `entryCase`, _however_ you cannot use `PlainEntrypointsKind` for both levels (or you'll see the issue that you described) because these are at the same "level" AFAIU.
The solution would be to define and use some other `*EntrypointKind`, for an example see `morley-ledgers` and `FA2EntrypointsKind`+`FA2SampleParameter` and its [autodoc result](https://gitlab.com/morley-framework/morley-ledgers/-/blob/autodoc/master/autodoc/FA2Sample.md#section-FA2EntrypointsKind).

However looking at that, I'd say that it's also not ideal (due to the TOC) even tho the "how to call" sections are just right.

So, I suggest keeping the current choice of having a single `Parameter` and I think this is another thing worthy of re-consideration in `morley` itself :thinking: